src/Components/Components/src/Reflection/ComponentProperties.cs

```diff
@@ -245,9 +246,15 @@ private static void ThrowForInvalidCaptureUnmatchedValuesParameterType(Type targ
 
         private class WritersForType
         {
+            private const int MaxCachedWriterLookups = 100;
+            private readonly Dictionary<string, IPropertySetter> _underlyingWriters;
+            private readonly ConcurrentDictionary<StringObjectIdentityKey, IPropertySetter?> _writerCache;
+
             public WritersForType(Type targetType)
             {
-                WritersByName = new Dictionary<string, IPropertySetter>(StringComparer.OrdinalIgnoreCase);
+                _underlyingWriters = new Dictionary<string, IPropertySetter>(StringComparer.OrdinalIgnoreCase);
+                _writerCache = new ConcurrentDictionary<StringObjectIdentityKey, IPropertySetter?>();
```

Comment:
There's a `ReferenceEqualityComparer<T>` type in 5.0 you could use. You can avoid create the type type

Reply:
Great, thanks for the tip! I switched over to this and not only does it save code, but it reduces the instruction count by another 1% on the ComplexTable benchmark (probably because it no longer has to call the constructor or field setter for my custom type).